src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,1072 @@
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            DumpEE();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractGainParameters();
+            ExtractTgcParameters();
+            ExtractResolutionParameters();
+            ExtractKsTaParameters();
+            ExtractKsToParameters();
+            ExtractCPParameters();
+            ExtractAlphaParameters();
+            ExtractOffsetParameters();
+            ExtractKtaPixelParameters();
+            ExtractKvPixelParameters();
+            ExtractCILCParameters();
+            ExtractDeviatingPixels();
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            ushort[] controlRegister1 = new ushort[1];
+            int value;
+
+            value = ((byte)mode & 0x07) << 7;
+
+            controlRegister1 = ReadRegisters16(0x800D, 1);
+            value = (controlRegister1[0] & 0xFC7F) | value;
+            Write16(0x800D, value);
+        }
+
+        /// <summary>
+        /// Request both 'halves' of a frame from the sensor, merge them
+        /// and calculate the temperature in C for each of 32x24 pixels.
+        /// </summary>
+        public float[] GetFrame()
+        {
+            float[] framebuf = new float[768];
+            float emissivity = 0.95f;
+            ushort[] mlx90640Frame = new ushort[834];
+
+            for (int i = 0; i < 2; i++)
+            {
+                if (GetFrameData(ref mlx90640Frame) < 0)
+                {
+                    throw new Exception("Frame data error");
+                }
+
+                float tr = GetTa(mlx90640Frame) - OPENAIR_TA_SHIFT;
+                CalculateTo(mlx90640Frame, emissivity, tr, ref framebuf);
+            }
+
+            return framebuf;
+        }
+
+        private void CalculateTo(ushort[] frameData, float emissivity, float tr, ref float[] result)
+        {
+            float vdd;
+            float ta;
+            float ta4;
+            float tr4;
+            float taTr;
+            float gain;
+            float[] irDataCP = new float[2];
+            float irData;
+            float alphaCompensated;
+            byte mode;
+            sbyte ilPattern;
+            byte chessPattern;
+            byte pattern;
+            byte conversionPattern;
+            float sx;
+            float to;
+            float[] alphaCorrR = new float[4];
+            byte range;
+            ushort subPage;
+            float ktaScale;
+            float kvScale;
+            float alphaScale;
+            float kta;
+            float kv;
+
+            subPage = frameData[833];
+            vdd = GetVdd(frameData);
+            ta = GetTa(frameData);
```

Comment:
if this returned Temperature instead you could use `.Kelvin` rather than adding 273.15